plugins/notifications-backend-module-slack/src/lib/SlackNotificationProcessor.ts

```diff
@@ -0,0 +1,285 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AuthService,
+  DiscoveryService,
+  LoggerService,
+} from '@backstage/backend-plugin-api';
+import { CatalogApi } from '@backstage/catalog-client';
+import { Entity, parseEntityRef } from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+import { NotFoundError } from '@backstage/errors';
+import { Notification } from '@backstage/plugin-notifications-common';
+import {
+  NotificationProcessor,
+  NotificationSendOptions,
+} from '@backstage/plugin-notifications-node';
+import { durationToMilliseconds } from '@backstage/types';
+import { Counter, metrics } from '@opentelemetry/api';
+import { ChatPostMessageArguments, WebClient } from '@slack/web-api';
+import DataLoader from 'dataloader';
+import pThrottle from 'p-throttle';
+import {
+  ANNOTATION_SLACK_CHANNEL_ID,
+  ANNOTATION_SLACK_CHANNEL_NAME,
+  ANNOTATION_SLACK_USER_ID,
+} from './constants';
+import { toChatPostMessageArgs } from './util';
+
+export class SlackNotificationProcessor implements NotificationProcessor {
+  private readonly logger: LoggerService;
+  private readonly catalog: CatalogApi;
+  private readonly auth: AuthService;
+  private readonly slack: WebClient;
+  private readonly sendNotifications;
+  private readonly messagesSent: Counter;
+  private readonly messagesFailed: Counter;
+
+  static fromConfig(
+    config: Config,
+    options: {
+      auth: AuthService;
+      discovery: DiscoveryService;
+      logger: LoggerService;
+      catalog: CatalogApi;
+      slack?: WebClient;
+    },
+  ): SlackNotificationProcessor[] {
+    const slackConfig =
+      config.getOptionalConfigArray('notifications.processors.slack') ?? [];
+    return slackConfig.map(c => {
+      const token = c.getString('token');
+      const slack = options.slack ?? new WebClient(token);
+      return new SlackNotificationProcessor({
+        slack,
+        ...options,
+      });
+    });
+  }
+
+  private constructor(options: {
+    slack: WebClient;
+    auth: AuthService;
+    discovery: DiscoveryService;
+    logger: LoggerService;
+    catalog: CatalogApi;
+  }) {
+    const { auth, catalog, logger, slack } = options;
+    this.logger = logger;
+    this.catalog = catalog;
+    this.auth = auth;
+    this.slack = slack;
+
+    const meter = metrics.getMeter('default');
+    this.messagesSent = meter.createCounter(
+      'notifications.processors.slack.sent.count',
```

Comment:
another possibility: `notifications.processors.slack.sent`, which for example in Grafana you might graph as `notifications_processors_slack_sent_total` (which is how the exporter amends it), and then you could have the label `result: 'ok'` (or `'failed'`). Doing that instead of two individual metrics, lets you nicely co-graph them and compute failure rate etc.

Up to you though